net-libs/grpc/grpc-1.12.1.ebuild

```diff
@@ -0,0 +1,154 @@
+
+REQUIRED_USE="
+	python? ( ${PYTHON_REQUIRED_USE} )
+	tools? ( python )
+"
+
+DEPEND="
+	>=dev-libs/openssl-1.0.2:0=[-bindist]
+	>=dev-libs/protobuf-3.5:=
+	dev-util/google-perftools
+	net-dns/c-ares:=
+	sys-libs/zlib:=
+	python? ( ${PYTHON_DEPS}
+		dev-python/coverage[${PYTHON_USEDEP}]
+		dev-python/cython[${PYTHON_USEDEP}]
+		>=dev-python/protobuf-python-3.5.1:=[${PYTHON_USEDEP}]
+		dev-python/six[${PYTHON_USEDEP}]
+		dev-python/wheel[${PYTHON_USEDEP}]
+		virtual/python-enum34[${PYTHON_USEDEP}]
+		virtual/python-futures[${PYTHON_USEDEP}]
+		doc? (
+			dev-python/sphinx[${PYTHON_USEDEP}]
+			dev-python/sphinx_rtd_theme[${PYTHON_USEDEP}]
+		)
+	)
+	systemtap? ( dev-util/systemtap )
+"
+
+RDEPEND="${DEPEND}"
+
+PATCHES=(
+	"${FILESDIR}/0001-grpc-1.11.0-Fix-cross-compiling.patch"
+	"${FILESDIR}/0002-grpc-1.3.0-Fix-unsecure-.pc-files.patch"
+	"${FILESDIR}/0003-grpc-1.3.0-Don-t-run-ldconfig.patch"
+	"${FILESDIR}/0004-grpc-1.11.0-fix-cpp-so-version.patch"
+	"${FILESDIR}/0005-grpc-1.11.0-pkgconfig-libdir.patch"
+	"${FILESDIR}/0006-grpc-1.12.1-allow-system-openssl.patch"
+	"${FILESDIR}/0007-grpc-1.12.1-allow-system-zlib.patch"
+	"${FILESDIR}/0008-grpc-1.12.1-allow-system-cares.patch"
+)
+
+src_prepare() {
+	sed -i 's@$(prefix)/lib@$(prefix)/$(INSTALL_LIBDIR)@g' Makefile || die
+	default
+	use python && distutils-r1_src_prepare
+}
+
+python_prepare() {
+	if use tools; then
+		rm -r third_party/protobuf || die
+		ln -s "${S}"/../protobuf-"${PROTOBUF_VERSION}" third_party/protobuf || die
+
+		pushd tools/distrib/python/grpcio_tools >/dev/null || die
+			ln -s ../../../../src || die
+			ln -s ../../../../third_party || die
+			ln -s ../../../../ grpc_root || die
+		popd >/dev/null || die
+	fi
+}
+
+src_compile() {
+	tc-export CC CXX PKG_CONFIG
+	emake \
+		V=1 \
+		prefix=/usr \
+		INSTALL_LIBDIR="$(get_libdir)" \
+		AR="$(tc-getAR)" \
+		AROPTS="rcs" \
+		CFLAGS="${CFLAGS}" \
+		LD="${CC}" \
+		LDXX="${CXX}" \
+		STRIP=true \
+		HOST_CC="$(tc-getBUILD_CC)" \
+		HOST_CXX="$(tc-getBUILD_CXX)" \
+		HOST_LD="$(tc-getBUILD_CC)" \
+		HOST_LDXX="$(tc-getBUILD_CXX)" \
+		HOST_AR="$(tc-getBUILD_AR)" \
+		HAS_SYSTEMTAP="$(usex systemtap true false)"
+
+	use python && distutils-r1_src_compile
+}
+
+python_compile() {
+	export GRPC_PYTHON_BUILD_SYSTEM_CARES=1
+	export GRPC_PYTHON_BUILD_SYSTEM_OPENSSL=1
+	export GRPC_PYTHON_BUILD_SYSTEM_ZLIB=1
+	export GRPC_PYTHON_BUILD_WITH_CYTHON=1
+	distutils-r1_python_compile
+
+	if use tools; then
+		pushd tools/distrib/python/grpcio_tools >/dev/null || die
+			distutils-r1_python_compile
+		popd >/dev/null || die
+	fi
+}
+
+python_compile_all() {
+	if use doc; then
+		esetup.py doc
+		mv doc/build doc/html || die
+	fi
+}
+
+src_install() {
+	emake \
+		prefix="${D}"/usr \
+		INSTALL_LIBDIR="$(get_libdir)" \
+		STRIP=true \
+		install
+
+	if use examples; then
+		docinto examples
+		dodoc -r examples/.
+		docompress -x /usr/share/doc/${PF}/examples
+	fi
+
+	use doc && local DOCS=( AUTHORS README.md doc/. )
```

Comment:
AUTHORS and README.md dont require those extra deps right? lets install those always then and only conditionally do the doc/ dir

Reply:
AUTHORS and README.md will be installed unconditionally.
because if I don't override DOCS, einstalldocs will use sane default value of DOCS and will include mentioned files. so it's all good. Just needed to mention those 2 files in USE=doc case.

Reply:
I actually tested it =)
now I have it installed and here are contents of docdir
```
/usr/share/doc/grpc-1.12.1/AUTHORS
/usr/share/doc/grpc-1.12.1/README.md.bz2
```

Reply:
oh okay, perfect :)